p2_GameOfLife.py

```diff
@@ -0,0 +1,56 @@
+# 289 Game of life
+
+# // Time Complexity : O(nxm)
+# // Space Complexity : O(1)
+# // Did this code successfully run on Leetcode : NO
+# // Any problem you faced while coding this : Yes, Not able to figure out the bug. The board is only updating "2" not "3" 
+
+
+# // code here along with comments explaining approach
+    # 1 -> 0 if neighbors<2
+    # 1 -> 1 if neighbors == 2 or 3
+    # 1 -> 0 if neighbors > 3
+    # 0 -> 1 if neighbors == 3
+
+    # 1->2 for 1->0 |  2 means previously it was 1
+    # 1->3 for 0->1 |  3 means previously it was 0
+
+
+
+class Solution:
+    def gameOfLife(self, board: List[List[int]]) -> None:
+        """
+        Do not return anything, modify board in-place instead.
+        """
+        
+        # countintg neighbours
+        def count_neighbours(board,n,m, row,col):
+            neighbors = 0
+            dir_arr = [[-1,-1],[-1,0],[-1,1],[0,-1],[0,1],[1,-1],[1,0],[1,1]]
+            for dir in dir_arr:
+                r = row+dir[0]
+                c = col+dir[1]
+                if r <=0 and c <=0 and r < n and c < m and (board[r][c]==1 or board[r][c]==2):
```

Comment:
r>=0 and c>=0

Reply:
Thanks! I didn't notice that bounds error I will update!